### PR TITLE
feat(discover): Add prototype impact score

### PR DESCRIPTION
**Pull request by dcramer**

### CODE REVIEW

File: src/sentry/api/event_search.py [hunk 1]
```diff
@@ -684,6 +684,12 @@ def to_list(value):
     # Long term these will become more complex functions but these are
     # field aliases.
     "apdex": {"result_type": "number", "aggregations": [["apdex(duration, 300)", "", "apdex"]]},
+    "impact": {
+        "result_type": "number",
+        "aggregations": [
+            ["(1 - apdex(duration, 300) + (1 - 1 / sqrt(uniq(user))) * 3", "", "impact"]
```
Comment (dashed) [inline src/sentry/api/event_search.py]: Looks like you're missing a closing `)` bracket somewhere in this expression.

Comment (dcramer) [reply src/sentry/api/event_search.py]: good catch

File: src/sentry/api/event_search.py [hunk 2]
```diff
@@ -684,6 +684,16 @@ def to_list(value):
     # Long term these will become more complex functions but these are
     # field aliases.
     "apdex": {"result_type": "number", "aggregations": [["apdex(duration, 300)", "", "apdex"]]},
+    "impact": {
+        "result_type": "number",
+        "aggregations": [
+            [
+                "(1 - ((countIf(duration < 300) + (countIf((duration > 300) AND (duration < 1200)) / 2)) / count())) + (1 - 1 / sqrt(uniq(user))) * 3",
```
Comment (dashed) [inline src/sentry/api/event_search.py]:
```suggestion
                "(1 - ((countIf(duration < 300) + (countIf((duration > 300) AND (duration < 1200)) / 2)) / count())) + ((1 - 1 / sqrt(uniq(user))) * 3)",
```

@dcramer just need some more parenthesis

**Review by dashed — CHANGES_REQUESTED** [review]
I was able to get it working, but just need to wrap the impact term with an extra set of parenthesis. 

Screen Shot 2019-11-21 at 5 56 10 PM

**Comment by dcramer** [comment]
> I was able to get it working, but just need to wrap the impact term with an extra set of parenthesis.
> 
> Screen Shot 2019-11-21 at 5 56 10 PM

i swear i just copy pasted what i ran in prod.. maybe it wasnt even working correctly in prod lol

**Review by dashed — APPROVED** [review]
Looks good to me 👍

File: src/sentry/static/sentry/app/views/eventsV2/eventQueryParams.tsx [hunk 3]
```diff
@@ -141,7 +141,7 @@ export const FIELDS: {[key: string]: ColumnValueType} = {
   'transaction.duration': 'duration',
   'transaction.op': 'string',
   apdex: 'number',
-
+  impact: 'number',
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/eventsV2/eventQueryParams.tsx]: If impact needs to be plotted on the graph it needs to be added to the AGGREGATE_FIELD_ALIAS list as well.

File: tests/sentry/api/test_event_search.py [hunk 4]
```diff
@@ -1036,12 +1036,17 @@ def test_automatic_fields_with_aggregate_aliases(self):
...
     def test_field_alias_duration_expansion(self):
-        fields = ["avg(transaction.duration)", "apdex", "p75", "p95", "p99"]
+        fields = ["avg(transaction.duration)", "apdex", "impact", "p75", "p95", "p99"]
         result = resolve_field_list(fields, {})
         assert result["selected_columns"] == []
         assert result["aggregations"] == [
             ["avg", "transaction.duration", "avg_transaction_duration"],
             ["apdex(duration, 300)", "", "apdex"],
+            [
+                "(1 - ((countIf(duration < 300) + (countIf((duration > 300) AND (duration < 1200)) / 2)) / count())) + (1 - 1 / sqrt(uniq(user))) * 3",
```
Comment (dashed) [inline tests/sentry/api/test_event_search.py]: @dcramer also need to update this

**Pull request merged**